recipes/repository.rb

```diff
@@ -25,6 +25,13 @@
     action :install
   end
 
+  # Trust new APT key
+  execute 'install new apt key' do
+    command 'apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 A2923DFF56EDA6E76E55E492D3A80E30382E94DE'
+    not_if 'apt-key list | grep 382E94DE'
+  end
+
```

Comment:
So I had another look at the `apt` cookbook, and unfortunately we can't really re-use anything from there, so I think I have to stick to the current approach.

About the rpm key: from what I've tested the `rpm --import`, `rpm -q` and `gpg --with-fingerprint <keyfile>` have very consistent behaviors (I've checked manually on CentOS 5 and 7). I understand your concern though, especially on the `grep` command which could be very sensitive to the slightest change in `gpg`'s output, I'll see if this could be made a little clearer/more resilient.